gateway-messages/src/sp_impl.rs

```diff
@@ -493,11 +506,31 @@ pub fn handle_message<H: SpHandler>(
             )
         }
         Some(OutgoingTrailingData::CabooseData(data)) => {
-            out[n..n + data.len()].copy_from_slice(data);
-            data.len()
+            // Because we didn't know the encoded length of the packet until
+            // now, we have to call back into the handler and copy caboose data
+            // into the trailing region.
+            //
+            // Unfortunately, unlike every other possible form of trailing data,
+            // this is faillible!  We check for errors here and re-serialize the
+            // whole chunk of packet data if there's a failure.
+            let len = data.len();
```

Comment:
Maybe worth adding to the comment that we know `out[n..][..len]` won't panic because we already checked that `len <= crate::MIN_TRAILING_DATA_LEN`.